cmd/lncli/commands.go

```diff
@@ -2117,12 +2137,15 @@ func updateChannelPolicy(ctx *cli.Context) error {
 
 	switch {
 	case ctx.IsSet("time_lock_delta"):
-		timeLockDelta = ctx.Int64("time_lock_delta")
+		timeLockDeltaStr := ctx.String("time_lock_delta")
```

Comment:
why are we parsing a `Uint64` flag as a string?

Reply:
when `"time_lock_delta"` is bigger than `max(int64)`, `ctx.Int64()` will return `0`:

```go
// Int64 looks up the value of a local Int64Flag, returns
// 0 if not found
func (c *Context) Int64(name string) int64 {
	return lookupInt64(name, c.flagSet)
}

func lookupInt64(name string, set *flag.FlagSet) int64 {
	f := set.Lookup(name)
	if f != nil {
		parsed, err := strconv.ParseInt(f.Value.String(), 0, 64)
		if err != nil {
			return 0
		}
		return parsed
	}
	return 0
}

```

Reply:
but then why bother with making the flat Uint64 at all? Why not string? and also , does this issue not just apply to all our Uint flags?

Reply:
> 

make the flag to `Uint64` to hint user the `time_lock_delta` is an unsigned integer.

Reply:
>  does this issue not just apply to all our Uint flags

yes, same for `ctx.Bool()` and `ctx.Float64()`, all of them have this issue.

Reply:
gotcha